core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -416,8 +416,9 @@ public void flushIfNeeded() throws IOException {
     }
 
     private boolean isGiantKey(Key k) {
-      // consider a key thats more than 3 standard deviations from previously seen key sizes as giant
-      return k.getSize() > keyLenStats.getMean() + keyLenStats.getStandardDeviation() * 3;
+      double mean = keyLenStats.getMean();
+      double stddev = keyLenStats.getStandardDeviation();
+      return k.getSize() > mean + 3 * Math.max(mean, stddev);
```

Comment:
Could add some comments here about why this logic is used to determine if key is giant...